lib/mongoid/scroll/base64_encoded_cursor.rb

```diff
@@ -4,46 +4,37 @@
 module Mongoid
   module Scroll
     # Allows to serializer/deserialize the cursor using RFC 4648
-    class Base64EncodedCursor < Cursor
-      class << self
-        def from_cursor(cursor)
-          base64_encoded_cursor = new(nil,
-                                      field_type: cursor.field_type,
-                                      field_name: cursor.field_name,
-                                      direction: cursor.direction,
-                                      include_current: cursor.include_current)
-          base64_encoded_cursor.value = cursor.value
-          base64_encoded_cursor.tiebreak_id = cursor.tiebreak_id
-          base64_encoded_cursor
+    class Base64EncodedCursor < BaseCursor
+      def initialize(value, options = {})
+        options = BaseCursor.extract_field_options(options)
+        if value
+          begin
+            parsed = ::JSON.parse(::Base64.strict_decode64(value))
+          rescue
+            raise Mongoid::Scroll::Errors::InvalidBase64CursorError.new(cursor: value)
+          end
+          super BaseCursor.parse_field_value(parsed['field_type'], parsed['field_name'], parsed['value']), {
+            field_type: BaseCursor.parse_field_type(parsed['field_type'], parsed['field_name']),
```

Comment:
`BaseCursor.parse_field_type` returns the class (e.g. `Date`) while `BaseCursor.extract_field_options` used in `Cursor` return the class as a string (e.g. `"Date"`).

With the class, the `scroll` method will raise a `MismatchedSortFieldsError` because the class will be compared with a string in `Scrollable::Fields#different_sort_fields?`. We could either use `to_s` in `BaseCursor#sort_options` or not use `BaseCursor.parse_field_type` here to do things like in `Cursor` (if so, the method will have to be deleted):
```suggestion
            field_type: parsed['field_type'],
```

```ruby
[3] pry(main)> c=nil; Feed::Item.asc(:name).limit(2).scroll(Mongoid::Scroll::Base64EncodedCursor) {|_, cr| c = cr}; c

[7] pry(main)> Feed::Item.asc(:name).limit(2).scroll(c).pluck(:id)
=> [BSON::ObjectId('6075ad853298c919807b4b52'), BSON::ObjectId('5e8d90cf5bd5441f2894f2e7')]

[8] pry(main)> Feed::Item.asc(:name).limit(2).scroll(Mongoid::Scroll::Base64EncodedCursor.new(c.to_s)).pluck(:id)
Mongoid::Scroll::Errors::MismatchedSortFieldsError:
```